### PR TITLE
Tab App Navigation Sample Feature Supports Teams T2.1

### PR DESCRIPTION
**Pull request by Mohammed-MSFT**

### CODE REVIEW

**Comment by Mohammed-MSFT** [comment]
Executed Build Pipeline : https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6947253254

File: samples/tab-app-navigation/nodejs/src/components/configure.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                    websiteUrl: `${window.location.origin}/default_tab`,
+                });
+
+                saveEvent.notifySuccess();
+            });
+
+            microsoftTeams.pages.config.setValidityState(true);
+        });
+    }, []);
+    return (
```
Comment (Pawank-MSFT) [inline samples/tab-app-navigation/nodejs/src/components/configure.jsx:30]: Give one line space.

**Review by Pawank-MSFT — APPROVED** [review]
One small comment please fix.

**Review by Prithvi-MSFT — APPROVED** [review]

File: samples/tab-app-navigation/nodejs/package.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "tabappnavigation",
+  "version": "0.1.0",
```
[the remark this answers is not in the document]
Comment (Mohammed-MSFT) [reply samples/tab-app-navigation/nodejs/package.json:3]: fixed

File: samples/tab-app-navigation/nodejs/src/components/index.css [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
...A8;%7D .st3%7Bfill:%23328BD8;%7D .st4%7Bfill:%23B6DCF1;%7D .st5%7Bopacity:0.2;fill:url(%23SVGID_1_);enable-background:new ;%7D%0A%3C/style%3E%3Crect y='1.1' class='st0' width='4640' height='646.3'/%3E%3Cpath class='st1' d='M3987.8,323.6L4310.3,1.1h-65.6l-460.1,460.1c-17.5,17.5-46.1,17.5-63.6,0L3260.9,1.1H0v646.3h3660.3 L3889,418.7c17.5-17.5,46.1-17.5,63.6,0l228.7,228.7h66.6l-260.2-260.2C3970.3,369.8,3970.3,341.1,3987.8,323.6z'/%3E%3Cpath class='st2' d='M3784.6,461.2L4244.7,1.1h-983.9l460.1,460.1C3738.4,478.7,3767.1,478.7,3784.6,461.2z'/%3E%3Cpath class='st3' d='M4640,1.1h-329.8l-322.5,322.5c-17.5,17.5-17.5,46.1,0,63.6l260.2,260.2H4640L4640,1.1L4640,1.1z'/%3E%3Cpath class='st4' d='M3889,418.8l-228.7,228.7h521.1l-228.7-228.7C3935.2,401.3,3906.5,401.3,3889,418.8z'/%3E%3ClinearGradient id='SVGID_1_' gradientUnits='userSpaceOnUse' x1='3713.7576' y1='438.1175' x2='3911.4084' y2='14.2535' gradientTransform='matrix(1 0 0 -1 0 641.3969)'%3E%3Cstop offset='0' style='stop-color:%23FFFFFF;stop-opacity:0.5'/%3E%3Cstop offset='1' style='stop-color:%23FFFFFF'/%3E%3C/linearGradient%3E%3Cpath class='st5' d='M3952.7,124.5c-17.5-17.5-46.1-17.5-63.6,0l-523,523h1109.6L3952.7,124.5z'/%3E%3C/svg%3E%0A");
```
[the remark this answers is not in the document]
Comment (Mohammed-MSFT) [reply samples/tab-app-navigation/nodejs/src/components/index.css:14]: Yes we are using

File: samples/tab-app-navigation/nodejs/src/components/index.css [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+  min-width: 580px;
+  max-width: 980px;
+  margin-left: auto;
+  margin-right: auto;
+  left: 0;
+  right: 0;
+  transition: bottom 400ms;
+}
+
+.ms-logo {
```
[the remark this answers is not in the document]
Comment (Mohammed-MSFT) [reply samples/tab-app-navigation/nodejs/src/components/index.css:201]: yes, we are using for landing page when we start running sample.

File: samples/tab-app-navigation/nodejs/src/components/tabOne.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        app.initialize().then(app.getContext).then((context: any) => {
+            if (pages.currentApp.isSupported()) {
+                const navPromise = pages.currentApp.navigateTo({ pageId: "tab_two", subPageId: "" });
+                navPromise.
+                    then((result) => console.log("Navigation Successfull", result)).
+                    catch((error) => console.log("Navigation Failed", error));
+            }
+            else {
+                const navPromise = pages.navigateToApp({
+                    appId: "", pageId: "tab_two"
```
[the remark this answers is not in the document]
Comment (Mohammed-MSFT) [reply samples/tab-app-navigation/nodejs/src/components/tabOne.tsx]: its an optional,

File: samples/tab-app-navigation/nodejs/src/components/tabTwo.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    const navigateBetweenTabs = () => {
+        app.initialize().then(app.getContext).then((context: any) => {
+            if (pages.currentApp.isSupported()) {
+                const navPromise = pages.currentApp.navigateTo({ pageId: "tab_three", subPageId: "" });
+                navPromise.
+                    then((result) => console.log("Navigation Successfull", result)).
+                    catch((error) => console.log("Navigation Failed", error));
+            }
+            else {
+                alert("Not Supported");
```
[the remark this answers is not in the document]
Comment (Mohammed-MSFT) [reply samples/tab-app-navigation/nodejs/src/components/tabTwo.tsx]: fixed

File: samples/tab-app-navigation/nodejs/src/components/configure.jsx (hunk 1 above)
Comment (Mohammed-MSFT) [reply samples/tab-app-navigation/nodejs/src/components/configure.jsx:30]: fixed

**Comment by Mohammed-MSFT** [comment]
Executed Build Pipeline : https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/7014351483

**Comment by ChetanSharma-msft** [comment]
@Mohammed-MSFT - Please check the feedback shared by developer and raise doc related issues for API's and as it's supported for only in Teams 2.1, so I asked doc team to update it here: https://github.com/MicrosoftDocs/msteams-docs/issues/8855

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**